src/js_results.hpp

```diff
@@ -196,7 +196,10 @@ typename T::Object ResultsClass<T>::create_filtered(ContextType ctx, const U &co
     NativeAccessor<T> accessor(ctx, realm, object_schema);
     query_builder::ArgumentConverter<ValueType, NativeAccessor<T>> converter(accessor, &args.value[1], args.count - 1);
     query_builder::apply_predicate(query, result.predicate, converter, mapping);
-    query_builder::apply_ordering(ordering, query.get_table(), result.ordering);
+    query_builder::apply_ordering(ordering, query.get_table(), result.ordering, mapping);
+    if (ordering.will_apply_include()) {
+        throw std::runtime_error("An INCLUDE clause is not supported on local queries, only on query based subscriptions.");
```

Comment:
Does this really need to throw an error? It seems reasonable to just ignore it so that doing something like grabbing a query off a subscription object and passing it to filtered() always works.

Reply:
The restriction is entirely arbitrary. During the sync implementation (https://github.com/realm/realm-sync/pull/2892#issuecomment-476427632) it was recommended that the js binding throws if this syntax is attempted on a local query because it is probably an error if someone tries to do this. A counter argument might be that users want to copy out the subscriptions as listed in their query based sync Realms on the server and run them verbatim locally? I don't know if that's a strong enough argument against saying that any other use of the "INCLUDE" syntax is invalid locally. It is intended to be a server side operation only. I don't have a strong opinion on it, just slightly leaning towards it being an error in local js queries because there is no way the js binding would generate something like this internally and so it is most likely a user error and having a descriptive error message is probably better than silently doing nothing.

Reply:
I mostly don't see any reason why someone would have an INCLUDE clause in their query other than grabbing it off a subscription. There's no reason for them to write one manually and we probably shouldn't even document it, so I don't think anyone would ever expect it to actually do something and then be surprised when it doesn't? As long as `.filtered('... INCLUDES (...)').subscribe()` actually does pass the includes clause to the subscription I don't think it really counts as silently doing nothing when the user has tried to ask for something. If it doesn't just happen to work then it's not worth making it work and should just throw, though.

Reply:
Note, that with new subscription management API's getting to the underlying query description is easier than it used to be. So I suspect that people might be copy/pasting it to e.g. Studio when debugging. I'm not sure if that is a strong enough argument for allowing it to run locally though.

Reply:
In any case, it is easier to be more restrictive from the start, and enough people actually run into this exception we can consider removing it?

Reply:
I lifted the restriction, so now queries with "INCLUDE" will do the right thing and added tests for it. However, I agree we shouldn't document it as officially part of the query syntax as it might be confusing.